### ISSUE TITLE
[macOS] Resolution problem/flickering

### ISSUE THREAD

**Issue by StenApp** [m64p/mupen64plus-gui#33]
compiled the mupen64plus-gui on Mac and getting this problem

sometimes the picture is smaller than the window

see here for pictures:
gonetz/GLideN64#1659

toggle fullscreen is broken too

**Comment by StenApp** [comment]
@loganmc10 

Do you have a clue what could cause this?
Thanks for the other quick answers

**Comment by loganmc10** [comment]
Sorry I meant to respond to this. After I saw this I re-tested Linux and Windows 10. I had no issues resizing, or going in and out of fullscreen. I haven't heard about any issues from others either.

There is no platform specific code (like there is no code saying "if Windows: do it this way, if Linux: do it that way", so I'm not sure why Mac OS would have an issue.

**Comment by StenApp** [comment]
it seems to give me a full window picture for a short period if screen changes e.g. select a mode or next screen is to be rendered (pause menu in game).
Am I able to debug this anyhow?

**Comment by loganmc10** [comment]
Yeah I'm really not sure, without a machine to test I can't even really guess what the problem might be. Does it happen with other games or just Vigilante 8?

**Comment by StenApp** [comment]
no, the same for Goldeneye E!, Rogue Squadron E!, Perfect Dark E!

**Comment by loganmc10** [comment]
Hm ok I tried a few E roms on Linux, still no issue.

There is no real debug build or anything.

The resizing is handled here by mupen64plus-gui:
https://github.com/m64p/mupen64plus-gui/blob/master/oglwindow.cpp#L27-L44

And here by GLideN64:
https://github.com/gonetz/GLideN64/blob/master/src/Graphics/OpenGLContext/mupen64plus/mupen64plus_DisplayWindow.cpp#L127-L153

You can try messing around with those functions a bit, besides that there isn't any verbose output or anything that would be any help

**Comment by StenApp** [comment]
I try Qt Creator, maybe that helps...
Thanks for the info

**Comment by StenApp** [comment]
I'm out of ideas:

Starting /Users/stenappelt/Downloads/mupen64plus-gui/mupen64plus-gui.app/Contents/MacOS/mupen64plus-gui...
dyld: Symbol not found: __cg_png_create_info_struct
  Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
  Expected in: /usr/local/lib/libPng.dylib
 in /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
Das Programm ist abgestürzt.
The process was ended forcefully.
/Users/stenappelt/Downloads/mupen64plus-gui/mupen64plus-gui.app/Contents/MacOS/mupen64plus-gui crashed.

I don't know how to fix.
Guess paths to libs are wrong, file is there.

**Comment by Joomla12** [comment]
What is your aspect ratio set to in Glide? If you have it set to "adjust" it'll cause problems like what you're describing.

**Comment by StenApp** [comment]
My aspect ratio is set to 1 in this Gui.

**Comment by Joomla12** [comment]
Do you have a Retina display? I wonder if that could be the cause. I have one, but I haven't tested it because my MacBook is indefinitely connected to my living room TV.

**Comment by StenApp** [comment]
Yes i have a macbook pro with retina display and a standalone Radeon.

**Comment by StenApp** [comment]
```
[gles2GlideN64]: _setAttributes
[gles2GlideN64]: Create setting videomode 800x600
OpenGL major version: 4
OpenGL minor version: 1
OpenGL vendor: ATI Technologies Inc.
OpenGL renderer: AMD Radeon Pro 560 OpenGL Engine
```
I'm out of ideas any hint is welcome

**Comment by StenApp** [comment]
Not solved but I may never work, thus closed

**Comment by StenApp** [comment]
Resize problem?
QT Problem
OpenGL Problem?

**Comment by loganmc10** [comment]
Mac problem, it works on Windows and Linux

**Comment by programWhiz** [comment]
Hey this may not fix your issue, but for me it was the rice driver.  Try using glide, it fixed my issues:
```
mupen64plus --gfx mupen64plus-video-glide64 <rom>
```
